### PR TITLE
enable product

### PR DESCRIPTION
**Pull request by Tigo9527**

### CODE REVIEW

File: src/main/java/conflux/dex/model/Product.java [hunk 1]
```diff
@@ -50,6 +50,8 @@ public class Product {
 	 * Minimum order funds. (For limit orders, funds = order.price * order.amount; for market-buy orders, funds = order.amount)
 	 */
 	private BigDecimal minOrderFunds;
+
+	private Boolean enable;
```
Comment (boqiu) [inline src/main/java/conflux/dex/model/Product.java]: use `bool` type for avoid null pointer?

File: src/main/resources/initdb.sql [hunk 2]
```diff
@@ -26,6 +26,7 @@ CREATE TABLE IF NOT EXISTS t_product (
 	min_order_amount DECIMAL(32,18) NOT NULL,
 	max_order_amount DECIMAL(32,18) NOT NULL,
 	min_order_funds DECIMAL(32,18) NOT NULL,
+    enable BOOLEAN NOT NULL DEFAULT TRUE,
```
Comment (boqiu) [inline src/main/resources/initdb.sql]: How about to rename to `disabled`, which is `false` by default.

File: src/main/java/conflux/dex/controller/ProductController.java [hunk 3]
```diff
@@ -269,11 +271,15 @@ public Product get(@PathVariable String nameOrId) {
 	 */
 	@GetMapping
 	public ProductPagingResult list(
-			@RequestParam(required = false, defaultValue = "0") int offset, 
+			@RequestParam(required = false, defaultValue = "true") Boolean filter,
```
Comment (boqiu) [inline src/main/java/conflux/dex/controller/ProductController.java]: Better to define the `filter` as last parameter, and filter out disabled products by default.

**Review by boqiu — CHANGES_REQUESTED** [review]

File: src/main/java/conflux/dex/model/Product.java (hunk 1 above)
Comment (Tigo9527) [reply src/main/java/conflux/dex/model/Product.java]: ok

File: src/main/resources/initdb.sql (hunk 2 above)
Comment (Tigo9527) [reply src/main/resources/initdb.sql]: ok

File: src/main/java/conflux/dex/controller/ProductController.java (hunk 3 above)
Comment (Tigo9527) [reply src/main/java/conflux/dex/controller/ProductController.java]: ok

File: src/main/java/conflux/dex/model/Product.java [hunk 4]
```diff
@@ -133,12 +133,12 @@ public void setMinOrderFunds(BigDecimal minOrderFunds) {
...
 
-	public Boolean getEnable() {
-		return enable;
+	public Boolean isDisabled() {
+		return disabled;
 	}
 
-	public void setEnable(Boolean enable) {
-		this.enable = enable;
+	public void setDisabled(Boolean disabled) {
```
Comment (boqiu) [inline src/main/java/conflux/dex/model/Product.java]: Use `boolean` type as well.

File: src/main/java/conflux/dex/model/Product.java [hunk 5]
```diff
@@ -133,12 +133,12 @@ public void setMinOrderFunds(BigDecimal minOrderFunds) {
 		this.minOrderFunds = minOrderFunds;
 	}
 
-	public Boolean getEnable() {
-		return enable;
+	public Boolean isDisabled() {
```
Comment (boqiu) [inline src/main/java/conflux/dex/model/Product.java]: Use  `boolean` type as well.

File: src/main/java/conflux/dex/controller/ProductController.java [hunk 6]
```diff
@@ -271,14 +271,14 @@ public Product get(@PathVariable String nameOrId) {
 	 */
 	@GetMapping
 	public ProductPagingResult list(
-			@RequestParam(required = false, defaultValue = "true") Boolean filter,
 			@RequestParam(required = false, defaultValue = "0") int offset,
-			@RequestParam(required = false, defaultValue = "10") int limit) {
+			@RequestParam(required = false, defaultValue = "10") int limit,
+			@RequestParam(required = false, defaultValue = "true") Boolean filter) {
```
Comment (boqiu) [inline src/main/java/conflux/dex/controller/ProductController.java]: Use `boolean` type?

**Review by boqiu — CHANGES_REQUESTED** [review]

File: src/main/java/conflux/dex/model/Product.java (hunk 4 above)
Comment (Tigo9527) [reply src/main/java/conflux/dex/model/Product.java]: ok

File: src/main/java/conflux/dex/model/Product.java (hunk 5 above)
Comment (Tigo9527) [reply src/main/java/conflux/dex/model/Product.java]: ok

File: src/main/java/conflux/dex/controller/ProductController.java (hunk 6 above)
Comment (Tigo9527) [reply src/main/java/conflux/dex/controller/ProductController.java]: ok

**Review by boqiu — APPROVED** [review]

**Pull request merged**